haystack/document_stores/pinecone.py

```diff
@@ -0,0 +1,682 @@
+
+        index = index or self.index
+        index = self._sanitize_index_name(index)
+
+        if not self.pinecone_indexes.get(index):
+            raise Exception(
+                f"Index named '{index}' does not exist. Try reinitializing PineconeDocumentStore() and running 'update_embeddings()' to create and populate an index."
+            )
+
+        if return_embedding is None:
+            return_embedding = self.return_embedding
+
+        query_emb = query_emb.reshape(1, -1).astype(np.float32)
+
+        if self.similarity == "cosine":
+            self.normalize_embedding(query_emb)
+
+        res = self.pinecone_indexes[index].query(query_emb.tolist(), top_k=top_k, include_values=True, filter=filters)
+
+        score_matrix = []
+        vector_id_matrix = []
+        for match in res["results"][0]["matches"]:
+            score_matrix.append(match["score"])
+            vector_id_matrix.append(match["id"])
+
+        documents = self.get_documents_by_vector_ids(vector_id_matrix, index=index)
+
+        # assign query score to each document
+        scores_for_vector_ids: Dict[str, float] = {str(v_id): s for v_id, s in zip(vector_id_matrix, score_matrix)}
+        for i, doc in enumerate(documents):
+            raw_score = scores_for_vector_ids[doc.id]
+            doc.score = self.finalize_raw_score(raw_score, self.similarity)
+
+            if return_embedding is True:
+                # get embedding from Pinecone response
+                doc.embedding = self.pinecone_indexes[index].reconstruct(int(doc.id))
+
+        return documents
+
+    def save(self):
+        """
+        Save index to the specified file, not implemented for PineconeDocumentStore.
+        """
+        raise NotImplementedError("save method not implemented for PineconeDocumentStore")
+
+    def _load_init_params_from_config(
+        self, index_path: Optional[Union[str, Path]] = None, config_path: Optional[Union[str, Path]] = None
+    ):
+        raise NotImplementedError("Load init params from config not implemented for Pinecone")
+
+    def _limit_check(self, top_k: str, include_values: Optional[bool] = None):
+        """
+        Confirms the top_k value does not exceed Pinecone vector database limits.
+        """
+        if include_values:
+            if top_k > self.top_k_limit_vectors:
+                raise Exception(
+                    f"PineconeDocumentStore allows requests of no more than {self.top_k_limit_vectors} records ",
+                    f"when returning embedding values. This request is attempting to return {top_k} records.",
+                )
+        else:
+            if top_k > self.top_k_limit:
+                raise Exception(
+                    f"PineconeDocumentStore allows requests of no more than {self.top_k_limit} records. ",
+                    f"This request is attempting to return {top_k} records.",
+                )
+
+    @classmethod
+    def load():
```

Comment:
`cls` argument is missing here.

Reply:
added in latest commit, is now `def load(cls)`